editors/sigil/Portfile

```diff
@@ -51,9 +38,48 @@ compiler.cxx_standard   2011
 configure.args          -DCMAKE_BUILD_WITH_INSTALL_RPATH=OFF \
                         -DSYSTEM_LIBS_REQUIRED=ON \
                         -DUSE_SYSTEM_LIBS=ON
+post-extract {
+    if {[variant_isset python38]} {
+        reinplace "s|PythonInterp 3.7|PythonInterp 3.8|g" ${worksrcpath}/CMakeLists.txt
+        reinplace "s|PythonLibs 3.7|PythonLibs 3.8|g" ${worksrcpath}/CMakeLists.txt
+    }
+}
 
-configure.env-append    PATH=${frameworks_dir}/Python.framework/Versions/${python_branch}/bin:$env(PATH)
+pre-configure {
+    if {![variant_isset python38] && ![variant_isset python37]} {
+        ui_error "Sigil requires Python, please use a Python variant."
+        return -code error
+    }
+}
 
 destroot {
     copy ${destroot.dir}/bin/Sigil.app ${destroot}${applications_dir}
 }
+
+proc python-depends {python_branch frameworks_dir PATH} {
+    set python_version [string map {. ""} ${python_branch}]
+    depends_lib-append  port:python${python_version} \
+                        port:py${python_version}-chardet \
+                        port:py${python_version}-cssselect \
+                        port:py${python_version}-cssutils \
+                        port:py${python_version}-html5lib \
+                        port:py${python_version}-lxml \
+                        port:py${python_version}-Pillow \
+                        port:py${python_version}-regex \
+                        port:py${python_version}-six
+    configure.env-append PATH=${frameworks_dir}/Python.framework/Versions/${python_branch}/bin:${PATH}
+}
+
+variant python38 conflicts python37 description {Use python38 during build} {
```

Comment:
Also, I should have mentioned this before. The descriptions should be proper English and not use Macports shortcut names. So Change the description:

`Use python 3.8 during build
`